### PR TITLE
Fix portal page creation

### PR DESCRIPTION
**Pull request by piofinn**
## 📥 Proposed changes

Use MDX for all documentation pages in Portal. This removes the need for several dependencies, creates a neater page structure in production, and enables richer layouts.

This also fixes a bug where the Portal build crashed on pages with a redefined path.

## ☑️ Submission checklist

-   [x] I have read the [CONTRIBUTING](https://github.com/fremtind/jokul/blob/master/CONTRIBUTING.md) doc
-   [x] `yarn build` works locally with my changes
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [x] I have added necessary documentation (if appropriate)

## 💬 Further comments

This should probably have been several commits instead of one large, so here's a summary of changes made:

* Switched to `gatsby-plugin-page-creator` for all documentation pages, inluding texts in portal
* Switched to `.mdx` format on all texts in portal
* Rewrote GraphQL queries to reflect new output from page creation
* Updated portal navigation

affects: @fremtind/portal

### CODE REVIEW

File: portal/src/components/Layout/ComponentLayout.tsx [hunk 1]
```diff
@@ -27,8 +27,12 @@ export const ComponentLayout = ({
     setLocation(location);
     return (
         <Layout location={location} title={title} isComponentPage>
-            <h1 className="jkl-title-large">{title}</h1>
-            <GitHubLinks react={react} scss={scss} />
+            {(scss || react) && (
+                <>
+                    <h1 className="jkl-title-large">{title}</h1>
```
Comment (lfbergee) [inline portal/src/components/Layout/ComponentLayout.tsx]: Så tittelen skal bare være der hvis du har lenker til github?

File: portal/src/components/Sidebar/Sidebar.tsx [hunk 2]
```diff
@@ -32,7 +28,7 @@ export function Sidebar() {
             links = getStartedDocPages;
             break;
         case "komponenter":
```
Comment (lfbergee) [inline portal/src/components/Sidebar/Sidebar.tsx]: Disse brukes flere steder, foreslår å konstantifisere eller bruke enum

File: portal/gatsby-config.js [hunk 3]
```diff
@@ -13,19 +13,19 @@ module.exports = {
             options: { implementation: require("sass") },
         },
         {
-            resolve: `gatsby-source-filesystem`,
+            resolve: "gatsby-plugin-page-creator",
```
Comment (lfbergee) [inline portal/gatsby-config.js]: fint å få ryddet opp i denne :)

File: portal/src/components/Layout/ComponentLayout.tsx (hunk 1 above)
Comment (piofinn) [reply portal/src/components/Layout/ComponentLayout.tsx]: Hehe, den er litt klønete. De andre sidene har overskriften i markdown, så der blir det dobbelt. Og tittelen på siden er ikke alltid lik overskriften. Men bør nok skrives om så det gir mening

File: portal/src/components/Sidebar/Sidebar.tsx (hunk 2 above)
Comment (piofinn) [reply portal/src/components/Sidebar/Sidebar.tsx]: Kan legge det inn i `useNavigationLinks` f.eks.

**Pull request merged**